queries/python/highlights.scm

```diff
@@ -183,35 +170,51 @@
   "yield"
 ] @keyword
 
-[ "as" "from" "import"] @include
+["as" "from" "import"] @include
 
-[ "if" "elif" "else" ] @conditional
+["if" "elif" "else"] @conditional
 
-[ "for" "while" "break" "continue" ] @repeat
+["for" "while" "break" "continue"] @repeat
 
-[ "(" ")" "[" "]" "{" "}"] @punctuation.bracket
+["(" ")" "[" "]" "{" "}"] @punctuation.bracket
 
 (interpolation
   "{" @punctuation.special
   "}" @punctuation.special) @embedded
 
-[ "," "." ":" (ellipsis) ] @punctuation.delimiter
+["," "." ":" (ellipsis)] @punctuation.delimiter
 
-; Class definitions
+;; Class definitions
 
 (class_definition
-  name: (identifier) @type)
+  name: (identifier) @type
+  body: (block
+          (function_definition
+            name: (identifier) @method)))
+
 (class_definition
   superclasses: (argument_list
-    (identifier) @type))
+                  (identifier) @type))
 
 ((class_definition
   body: (block
           (expression_statement
             (assignment
               left: (expression_list
                       (identifier) @field)))))
- (match? @field "^([A-Z])@!.*$"))
+ (#match? @field "^([A-Z])@!.*$"))
+
+((class_definition
+  (block
+    (function_definition
+      name: (identifier) @constructor)))
+ (#match? @constructor "(__new__|__init__)"))
+
+; First parameter of a method is self or cls.
+(class_definition
+  body: (block
+          (function_definition
+            parameters: (parameters . (identifier) @constant.builtin))))
```

Comment:
Hmm, maybe that's better.

Reply:
Changed to match obj, cls and self. Looks like there is a bug though. The query doesn't work here, but it works on https://tree-sitter.github.io/tree-sitter/playground. Matching only one word works, but when matching different with `|` it doesn't work.


this doesn't work

```
((class_definition
  body: (block
          (function_definition
            parameters: (parameters . (identifier) @constant.builtin))))
 (#match? @constant.builtin "^(self|obj|cls)$"))
```

This works

```
((class_definition
  body: (block
          (function_definition
            parameters: (parameters . (identifier) @constant.builtin))))
 (#match? @constant.builtin "^cls$"))
```

Reply:
Looks like this is happening in all queries using `#match?`

Reply:
As a workaround you can try vim-match for old behavior

Reply:
Changed to vim-match